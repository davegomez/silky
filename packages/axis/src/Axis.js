// @flow
import React from 'react';
import type { CSS } from '../../utils/types';
import { Group } from '@silky/group';

type Props = {
  className?: string,
  css?: CSS,
  position: string,
};

export default function Text({ className = '', css, position }: Props) {
  return (
    <Group className={`silky-axis axis-${position} ${className}`} css={css}>
      {/* Test */}
    </Group>
  );
}
