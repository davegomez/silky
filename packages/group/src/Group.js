// @flow
import React from 'react';
import { getTransform } from '../../utils';
import type { CSS, Transform } from '../../utils/types';

type Props = {
  children: any,
  className?: string,
  css?: CSS,
  transform?: Transform,
};

export default function Group({
  children,
  className = '',
  css,
  transform,
  ...restProps
}: Props) {
  return (
    <g
      className={`silky-group ${className}`}
      css={css}
      transform={getTransform(transform)}
      {...restProps}
    >
      {children}
    </g>
  );
}
