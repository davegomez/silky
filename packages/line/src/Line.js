// @flow
import React from 'react';
import type { CSS } from '../../utils/types';

type Props = {
  className?: string,
  css?: CSS,
  x1: number | string,
  x2?: number | string,
  y1: number | string,
  y2?: number | string,
  pathLength?: number,
};

export default function Line({
  className = '',
  css,
  x1,
  x2,
  y1,
  y2,
  pathLength,
  ...restProps
}: Props) {
  return (
    <line
      className={`silky-line ${className}`}
      css={css}
      x1={x1}
      x2={x2}
      y1={y1}
      y2={y2}
      pathLength={pathLength}
      {...restProps}
    />
  );
}
