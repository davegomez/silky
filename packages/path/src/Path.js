// @flow
import React from 'react';
import type { CSS } from '../../utils/types';

type Props = {
  className?: string,
  css?: CSS,
  d: string,
  pathLength?: number,
};

export default function Path({
  className = '',
  css,
  d,
  pathLength,
  ...restProps
}: Props) {
  return (
    <path
      className={`silky-path ${className}`}
      css={css}
      d={d}
      pathLength={pathLength}
      {...restProps}
    />
  );
}
