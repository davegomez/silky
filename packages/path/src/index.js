export { default as Path } from './Path';
