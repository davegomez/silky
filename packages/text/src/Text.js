// @flow
import React from 'react';
import { getTransform } from '../../utils';
import type { CSS, Transform } from '../../utils/types';

type Props = {
  children: any,
  className?: string,
  css?: CSS,
  dx?: number | string,
  dy?: number | string,
  lengthAdjust?: 'spacing' | 'spacingAndGlyphs',
  transform?: Transform,
  textLength?: number | string,
  x?: number | string,
  y?: number | string,
};

export default function Text({
  children,
  className = '',
  css,
  dx,
  dy,
  lengthAdjust,
  transform,
  textLength,
  x,
  y,
  ...restProps
}: Props) {
  return (
    <text
      className={`silky-text ${className}`}
      css={css}
      dx={dx}
      dy={dy}
      lengthAdjust={lengthAdjust}
      textLength={textLength}
      transform={getTransform(transform)}
      x={x}
      y={y}
      {...restProps}
    >
      {children}
    </text>
  );
}
