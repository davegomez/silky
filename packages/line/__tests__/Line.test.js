import React from 'react';
import serializer, { matchers } from 'jest-emotion';
import { render, cleanup } from 'react-testing-library';
import { css } from '@emotion/core';
import { Line } from '../src';

expect.addSnapshotSerializer(serializer);
expect.extend(matchers);
afterEach(cleanup);

test('Line', () => {
  const { container } = render(
    <svg>
      <Line x1={0} y1={80} x2={100} y2={20} stroke="#AABBCC" />
    </svg>
  );

  const line = container.querySelector('line');
  expect(line).toMatchSnapshot();
  expect(line.classList.contains('silky-line')).toBe(true);
  expect(line.getAttribute('x1')).toBe('0');
  expect(line.getAttribute('x2')).toBe('100');
  expect(line.getAttribute('y1')).toBe('80');
  expect(line.getAttribute('y2')).toBe('20');
  expect(line.getAttribute('stroke')).toBe('#AABBCC');
});

test('Line with custom class', () => {
  const { container } = render(
    <svg>
      <Line
        x1="10%"
        y1="10%"
        x2="80%"
        y2="80%"
        stroke="black"
        className="foo"
      />
    </svg>
  );

  const line = container.querySelector('line');
  expect(line).toMatchSnapshot();
  expect(line.classList.contains('silky-line')).toBe(true);
  expect(line.getAttribute('x1')).toBe('10%');
  expect(line.getAttribute('x2')).toBe('80%');
  expect(line.getAttribute('y1')).toBe('10%');
  expect(line.getAttribute('y2')).toBe('80%');
  expect(line.getAttribute('stroke')).toBe('black');
  expect(line.classList.contains('foo')).toBe(true);
});

test('Line with Emotion', () => {
  const styles = css`
    stroke: black;
  `;
  const { container } = render(
    <svg>
      <Line x1={0} y1={80} x2={100} y2={20} css={styles} />
    </svg>
  );

  const line = container.querySelector('line');
  expect(line).toMatchSnapshot();
  expect(line.classList.contains('silky-line')).toBe(true);
  expect(line).toHaveStyleRule('stroke', 'black');
});

test('Line with pathLength', () => {
  const { container } = render(
    <svg>
      <Line x1="10%" y1="10%" x2="80%" y2="80%" stroke="black" pathLength={2} />
    </svg>
  );

  const line = container.querySelector('line');
  expect(line).toMatchSnapshot();
  expect(line.classList.contains('silky-line')).toBe(true);
  expect(line.getAttribute('pathLength')).toBe('2');
});
