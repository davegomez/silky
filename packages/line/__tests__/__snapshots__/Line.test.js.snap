// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Line 1`] = `
<line
  class="silky-line "
  stroke="#AABBCC"
  x1="0"
  x2="100"
  y1="80"
  y2="20"
/>
`;

exports[`Line with Emotion 1`] = `
.emotion-0 {
  stroke: black;
}

<line
  class="silky-line emotion-0"
  x1="0"
  x2="100"
  y1="80"
  y2="20"
/>
`;

exports[`Line with custom class 1`] = `
<line
  class="silky-line foo"
  stroke="black"
  x1="10%"
  x2="80%"
  y1="10%"
  y2="80%"
/>
`;

exports[`Line with pathLength 1`] = `
<line
  class="silky-line "
  pathLength="2"
  stroke="black"
  x1="10%"
  x2="80%"
  y1="10%"
  y2="80%"
/>
`;
